src/video/x11/SDL_x11events.c

```diff
@@ -1022,8 +1023,11 @@ X11_DispatchEvent(_THIS, XEvent *xevent)
             SDL_zeroa(text);
 #ifdef X_HAVE_UTF8_STRING
             if (data->ic) {
-                X11_Xutf8LookupString(data->ic, &xevent->xkey, text, sizeof(text),
+                int bytes = X11_Xutf8LookupString(data->ic, &xevent->xkey, text, sizeof(text),
                                   &keysym, &status);
+                if (bytes <= 0) {
```

Comment:
This might need to be a separate commit

Reply:
I updated this part to only use Xutf8LookpString for KeyPress, to match the man page of Xutf8LookpString.
```
Note: To ensure proper input processing, it is essential that the client pass only KeyPress events to  XmbLookupString,  XwcLookupString  and  Xutf8LookupString.  Their behavior when a client passes a KeyRelease event is undefined.
```